src/hist/basehist.py

```diff
@@ -282,6 +282,40 @@ def __setitem__(  # type: ignore
 
         return super().__setitem__(self._index_transform(index), value)
 
+    def profile(self: T, axis: Union[int, str]) -> T:
+        """
+        Returns a profile (Mean/WeightedMean) histogram from a normal histogram
+        with N-1 axes. The axis given is profiled over and removed from the
+        final histogram.
+        """
+
+        if self.kind != bh.Kind.COUNT:
+            raise TypeError("Profile requires a COUNT histogram")
+
+        axes = list(self.axes)
+        iaxis = axis if isinstance(axis, int) else self._name_to_index(axis)
+        axes.pop(iaxis)
+
+        values = self.values()
+        tmp_variances = self.variances()
+        variances = tmp_variances if tmp_variances is not None else values
+        centers = self.axes[iaxis].centers
+
+        count = np.sum(values, axis=iaxis)
+
+        num = np.tensordot(values, centers, ([iaxis], [0]))
+        num_err = np.sqrt(np.tensordot(variances, centers ** 2, ([iaxis], [0])))
+
+        den = np.sum(values, axis=axis)
+        den_err = np.sqrt(np.sum(variances, axis=axis))
+
+        new_values = num / den
+        new_variances = (num_err / den) ** 2 + (den_err * num / den ** 2) ** 2
```

Comment:
After checking consistency between ROOT and numpy, the error bars were not quite right. I updated https://github.com/scikit-hep/hist/issues/156 just now to fix the sign that I messed up, . This line should be
```python
new_variances = (num_err / den) ** 2 - (den_err * num / den ** 2) ** 2
```
(sign flipped in the middle)
Sorry...

Reply:
```suggestion
        new_variances = (num_err / den) ** 2 - (den_err * num / den ** 2) ** 2
```